src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapEmail.scala

```diff
@@ -25,4 +25,27 @@ object JmapEmail {
 
   val nonEmptyListMessagesChecks: HttpCheck =
     jsonPath("$.methodResponses[0][1].ids[*]").findAll.saveAs("emailIds")
+
+  def getRandomEmails(queryParameters: JmapParameters = NO_PARAMETERS): HttpRequestBuilder = {
```

Comment:
I bet you do not get a 'single' random email, but rather get them all...

To select a random email please stick to what is done on draft ():

 - 1. Do a EmailQuery first and save the returned Ids in the session (emailIds variable) - please do that as a separate spec as well - it enable a finer grain diagnostic.
 - 2. Then use the same syntax as `JmapMessages::getRandomMessages`: `"ids": ["$${$emailIds.random()}"],`

Reply:
ok with your comment I finally understood how it works... The query part is done in `CommonSteps.provisionUsersWithMessageList`on draft, I missed that one thanks...

Reply:
I would be in favor of moving the query part out of the provisionning. 

To make it explicit...